lib/pact/consumer_contract/pact_file.rb

```diff
@@ -36,39 +36,21 @@ def save_pactfile_to_tmp pact, name
     end
 
     def render_pact(uri_string, options)
-      uri_obj = URI(windows_safe(uri_string))
-      if uri_obj.userinfo
-        options[:username] = uri_obj.user unless options[:username]
-        options[:password] = uri_obj.password unless options[:password]
-      end
-      get(uri_obj, options)
+      uri_string.start_with?("http://", "https://") ? get_remote_with_retry(uri_string, options) : get_local(uri_string, options)
     end
 
     private
 
-    def get(uri, options)
-      local?(uri) ?  get_local(uri, options) : get_remote_with_retry(uri, options)
-    end
-
-    def local? uri
-      !uri.host
-    end
-
-    def get_local(uri, _)
-      File.read uri.to_s
-    end
-
-    def get_remote(uri, options)
-      request = Net::HTTP::Get.new(uri)
-      request.basic_auth(options[:username], options[:password]) if options[:username]
-      Net::HTTP.start(uri.host, uri.port, use_ssl: uri.scheme == 'https') do |http|
-        http.open_timeout = options[:open_timeout] || OPEN_TIMEOUT
-        http.read_timeout = options[:read_timeout] || READ_TIMEOUT
-        http.request(request)
-      end
+    def get_local(filepath, _)
+      File.read filepath
```

Comment:
Shouldn't this be `windows_safe(filepath)`?

Reply:
yep, you're right.